nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/test/java/org/apache/nifi/parameter/TestOnePasswordParameterProvider.java

```diff
@@ -0,0 +1,131 @@
+
+        // list items
+        final HttpRequestBodySpec bodySpecItems = mock(HttpRequestBodySpec.class);
+        when(uriSpec.uri(argThat(argument -> argument == null || argument.getPath().endsWith("/items")))).thenReturn(bodySpecItems);
+        when(bodySpecItems.header(any(), any())).thenReturn(bodySpecItems);
+        final HttpResponseEntity httpEntityItems = mock(HttpResponseEntity.class);
+        when(bodySpecItems.retrieve()).thenReturn(httpEntityItems);
+
+        final String responseBodyItems = "[{\"additionalInformation\":\"localhost\",\"category\":\"DATABASE\",\"createdAt\":\"2023-10-13T16:54:45Z\","
+                + "\"id\":\"evsdsvep67jitka2hmbg5tbxry\",\"lastEditedBy\":\"DBJG7MEI5NH6BIUVR3S66OWNSE\",\"title\":\"POSTGRES\","
+                + "\"updatedAt\":\"2023-10-13T16:55:47Z\",\"vault\":{\"id\":\"qeo4jajm7azfh3wnsynbmr5lem\",\"name\":\"Engineering\"},\"version\":2}]";
+        InputStream responseItems = new ByteArrayInputStream(responseBodyItems.getBytes());
+        when(httpEntityItems.body()).thenReturn(responseItems);
+
+        // get item
+        final HttpRequestBodySpec bodySpecItem = mock(HttpRequestBodySpec.class);
+        when(uriSpec.uri(argThat(argument -> argument == null || argument.getPath().endsWith("evsdsvep67jitka2hmbg5tbxry")))).thenReturn(bodySpecItem);
+        when(bodySpecItem.header(any(), any())).thenReturn(bodySpecItem);
+        final HttpResponseEntity httpEntityItem = mock(HttpResponseEntity.class);
+        when(bodySpecItem.retrieve()).thenReturn(httpEntityItem);
+
+        final String responseBodyItem = "{\"additionalInformation\":\"localhost\",\"category\":\"DATABASE\",\"createdAt\":\"2023-10-13T16:54:45Z\","
+                + "\"fields\":[{\"id\":\"notesPlain\",\"label\":\"notesPlain\",\"purpose\":\"NOTES\",\"type\":\"STRING\"},{\"id\":\"database_type\","
+                + "\"label\":\"Type\",\"type\":\"MENU\",\"value\":\"postgresql\"},{\"id\":\"hostname\",\"label\":\"serveur\",\"type\":\"STRING\","
+                + "\"value\":\"localhost\"},{\"id\":\"port\",\"label\":\"Port\",\"type\":\"STRING\",\"value\":\"5432\"},{\"id\":\"database\","
+                + "\"label\":\"Base de données\",\"type\":\"STRING\",\"value\":\"mydatabase\"},{\"id\":\"username\",\"label\":\"Nom d'utilisateur\","
+                + "\"type\":\"STRING\",\"value\":\"postgres\"},{\"id\":\"password\",\"label\":\"mot de passe\",\"type\":\"CONCEALED\","
+                + "\"value\":\"thisisabadpassword\"},{\"id\":\"sid\",\"label\":\"SID\",\"type\":\"STRING\"},{\"id\":\"alias\",\"label\":\"Alias\","
+                + "\"type\":\"STRING\"},{\"id\":\"options\",\"label\":\"Options de connexion\",\"type\":\"STRING\"}],\"id\":\"evsdsvep67jitka2hmbg5tbxry\","
+                + "\"lastEditedBy\":\"DBJG7MEI5NH6BIUVR3S66OWNSE\",\"title\":\"POSTGRES\",\"updatedAt\":\"2023-10-13T16:55:47Z\","
+                + "\"vault\":{\"id\":\"qeo4jajm7azfh3wnsynbmr5lem\",\"name\":\"Engineering\"},\"version\":2}";
```

Comment:
Is it necessary to include all of these elements? Although this mirrors what 1Password returns, it would be easier to maintain if it included only the fields that the service uses. As this is also rather long, this is a great opportunity to using the newer [String text block](https://www.baeldung.com/java-text-blocks) language feature.